ios/sound.mm

```diff
@@ -24,209 +24,323 @@
+        iCoreAudioBufferSizeStereo = 2 * iCoreAudioBufferSizeMono;
+
+        // create memory for intermediate audio buffer
+        vecsTmpAudioSndCrdStereo.Init ( iCoreAudioBufferSizeStereo );
+
+        AVAudioSession* sessionInstance = [AVAudioSession sharedInstance];
+
+        // we are going to play and record so we pick that category
+        NSError* error = nil;
+        [sessionInstance setCategory:AVAudioSessionCategoryPlayAndRecord
+                         withOptions:AVAudioSessionCategoryOptionMixWithOthers | AVAudioSessionCategoryOptionDefaultToSpeaker |
+                                     AVAudioSessionCategoryOptionAllowBluetooth | AVAudioSessionCategoryOptionAllowBluetoothA2DP |
+                                     AVAudioSessionCategoryOptionAllowAirPlay
+                               error:&error];
+
+        // using values from jamulus settings 64 = 2.67ms/2
+        NSTimeInterval bufferDuration = iCoreAudioBufferSizeMono / Float32 ( SYSTEM_SAMPLE_RATE_HZ ); // yeah it's math
+        [sessionInstance setPreferredIOBufferDuration:bufferDuration error:&error];
+
+        // set the session's sample rate 48000 - the only supported by Jamulus
+        [sessionInstance setPreferredSampleRate:SYSTEM_SAMPLE_RATE_HZ error:&error];
+        [[AVAudioSession sharedInstance] setActive:YES error:&error];
+
+        OSStatus status;
+
+        // Describe audio component
+        AudioComponentDescription desc;
+        desc.componentType         = kAudioUnitType_Output;
+        desc.componentSubType      = kAudioUnitSubType_RemoteIO;
+        desc.componentFlags        = 0;
+        desc.componentFlagsMask    = 0;
+        desc.componentManufacturer = kAudioUnitManufacturer_Apple;
+
+        // Get component
+        AudioComponent inputComponent = AudioComponentFindNext ( NULL, &desc );
+
+        // Get audio units
+        status = AudioComponentInstanceNew ( inputComponent, &audioUnit );
+        checkStatus ( status );
+
+        // Enable IO for recording
+        UInt32 flag = 1;
+        status      = AudioUnitSetProperty ( audioUnit, kAudioOutputUnitProperty_EnableIO, kAudioUnitScope_Input, kInputBus, &flag, sizeof ( flag ) );
+        checkStatus ( status );
+
+        // Enable IO for playback
+        status = AudioUnitSetProperty ( audioUnit, kAudioOutputUnitProperty_EnableIO, kAudioUnitScope_Output, kOutputBus, &flag, sizeof ( flag ) );
+        checkStatus ( status );
+
+        // Describe format
+        AudioStreamBasicDescription audioFormat;
+        audioFormat.mSampleRate       = SYSTEM_SAMPLE_RATE_HZ;
+        audioFormat.mFormatID         = kAudioFormatLinearPCM;
+        audioFormat.mFormatFlags      = kAudioFormatFlagsNativeEndian | kAudioFormatFlagIsFloat | kAudioFormatFlagIsPacked;
+        audioFormat.mFramesPerPacket  = 1;
+        audioFormat.mChannelsPerFrame = 2;  // steoreo, so 2 interleaved channels
+        audioFormat.mBitsPerChannel   = 32; // sizeof float32
```

Comment:
Again, if we're processing PCM 16 bit fixed, what needs changing here?  (And anywhere else in `audioFormat`.)